iris-mpc-cpu/src/hnsw/graph/graph_store.rs

```diff
@@ -0,0 +1,399 @@
+
+    /// A test database. It creates a unique schema for each test. Call
+    /// `cleanup` at the end of the test.
+    ///
+    /// Access the database with `&graph` or `graph.owned()`.
+    pub struct TestGraphPg<V: VectorStore> {
+        graph:       GraphPg<V>,
+        schema_name: String,
+    }
+
+    impl<V: VectorStore> TestGraphPg<V> {
+        pub async fn new() -> Result<Self> {
+            let schema_name = temporary_name();
+            let graph = GraphPg::new(&test_db_url()?, &schema_name).await?;
+            Ok(TestGraphPg { graph, schema_name })
+        }
+
+        pub async fn cleanup(&self) -> Result<()> {
+            cleanup(&self.graph.pool, &self.schema_name).await
+        }
+
+        pub fn owned(&self) -> GraphPg<V> {
+            GraphPg {
+                pool:    self.graph.pool.clone(),
+                phantom: PhantomData,
+            }
+        }
+    }
+
+    impl<V: VectorStore> Deref for TestGraphPg<V> {
+        type Target = GraphPg<V>;
+        fn deref(&self) -> &Self::Target {
+            &self.graph
+        }
+    }
+
+    impl<V: VectorStore> DerefMut for TestGraphPg<V> {
+        fn deref_mut(&mut self) -> &mut Self::Target {
+            &mut self.graph
+        }
+    }
+
+    fn test_db_url() -> Result<String> {
+        dotenvy::from_filename(DOTENV_TEST)?;
+        Ok(env::var(ENV_DB_URL)?)
+    }
+
+    fn temporary_name() -> String {
+        format!("{}_{}", SCHEMA_PREFIX, rand::random::<u32>())
+    }
+
+    async fn cleanup(pool: &sqlx::PgPool, schema_name: &str) -> Result<()> {
+        assert!(schema_name.starts_with(SCHEMA_PREFIX));
+        sqlx::query(&format!("DROP SCHEMA \"{}\" CASCADE", schema_name))
+            .execute(pool)
+            .await?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+#[cfg(feature = "db_dependent")]
+mod tests {
+    use super::{test_utils::TestGraphPg, *};
+    use crate::{
+        hawkers::plaintext_store::PlaintextStore,
+        hnsw::{GraphMem, HnswSearcher, SortedNeighborhood},
+    };
+    use aes_prng::AesRng;
+    use iris_mpc_common::iris_db::db::IrisDB;
+    use rand::SeedableRng;
+    use tokio;
+
+    #[tokio::test]
+    async fn test_db() {
+        let mut graph = TestGraphPg::<PlaintextStore>::new().await.unwrap();
+        let mut vector_store = PlaintextStore::new();
+        let rng = &mut AesRng::seed_from_u64(0_u64);
+
+        let vectors = {
+            let mut v = vec![];
+            for raw_query in IrisDB::new_random_rng(10, rng).db {
+                let q = vector_store.prepare_query(raw_query);
+                v.push(vector_store.insert(&q).await);
+            }
+            v
+        };
+
+        let distances = {
+            let mut d = vec![];
+            for v in vectors.iter() {
+                d.push(vector_store.eval_distance(&vectors[0], v).await);
+            }
+            d
+        };
+
+        let ep = graph.get_entry_point().await;
+
+        let ep2 = EntryPoint {
```

Comment:
Shouldn't you compare the content of `ep` with `ep2` as well?

Reply:
`ep` should be None so I added an `assert`.